File: LeetCode/LeetCode 1513. 仅含 1 的子串数.cpp
class Solution {
public:
    typedef long long ll;
    const int mod = 1e9 + 7;
    int numSub(string s) {
        ll len = 0, ans = 0;
        for(auto &x : s) {
            if(x == '1') len++;
            else {
                if(len & 1) ans = (ans + (1 + len) / 2 * len) % mod;
                else ans = (ans + len / 2 * (1 + len)) % mod;
                len = 0;
            }
        }
        if(len & 1) ans = (ans + (1 + len) / 2 * len) % mod;
        else ans = (ans + len / 2 * (1 + len)) % mod;
        return ans;
    }
};

File: LeetCode/LeetCode 1513. 仅含 1 的子串数.py
class Solution:
    def numSub(self, s: str) -> int:
        len, ans, mod = 0, 0, 1e9 + 7
        for i in s:
            if i == '1':
                len += 1
            else:
                ans = (ans + (1 + len) * len / 2) % mod
                len = 0 
        
        ans = (ans + (1 + len) * len / 2) % mod
        return int(ans)